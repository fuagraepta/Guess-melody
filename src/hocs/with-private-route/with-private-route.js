import React from 'react';
import PropTypes from 'prop-types';
import {Redirect} from 'react-router-dom';
import {compose} from 'recompose';
import {connect} from 'react-redux';

import {getAuthorizationStatus} from '../../reducer/user/selectors.js';

const withPrivateRoute = (Component) => {
  const WithPrivateRoute = (props) => {
    const {isAuthorizationRequired} = props;

    return (isAuthorizationRequired) ?
      <Redirect to = "/sign-in" />
      :
      <Component {...props} />;
  };

  WithPrivateRoute.propTypes = {
    isAuthorizationRequired: PropTypes.bool.isRequired
  };

  return WithPrivateRoute;
};


const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  isAuthorizationRequired: getAuthorizationStatus(state)
});

export {withPrivateRoute};
export default compose(connect(mapStateToProps), withPrivateRoute);
