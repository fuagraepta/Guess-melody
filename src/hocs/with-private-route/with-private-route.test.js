import React from 'react';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';

import {withPrivateRoute} from './with-private-route.js';

const MockComponent = () => {
  return (
    <div>
      Hello User!
    </div>
  );
};

const MockComponentWrapped = withPrivateRoute(MockComponent);

describe(`HOC with private route is rendered correctly.`, () => {
  it(`Component correctly renders if user sign in.`, () => {
    const mockComponentWrapped = renderer
      .create(
          <MockComponentWrapped
            isAuthorizationRequired = {false}
          />
      )
        .toJSON();
    expect(mockComponentWrapped).toMatchSnapshot();
  });

  it(`Component should not rendered if user sign out.`, () => {
    const mockComponentWrapped = renderer
      .create(
          <BrowserRouter>
            <MockComponentWrapped
              isAuthorizationRequired = {true}
            />
          </BrowserRouter>
      )
        .toJSON();
    expect(mockComponentWrapped).toMatchSnapshot();
  });
});
