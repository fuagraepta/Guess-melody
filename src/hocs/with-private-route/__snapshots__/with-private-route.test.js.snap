// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HOC with private route is rendered correctly. Component correctly renders if user sign in. 1`] = `
<div>
  Hello User!
</div>
`;

exports[`HOC with private route is rendered correctly. Component should not rendered if user sign out. 1`] = `null`;
